src/rules/jsxNoLambdaRule.ts

```diff
@@ -28,23 +29,34 @@ export class Rule extends Lint.Rules.AbstractRule {
 }
 
 class JsxNoLambdaWalker extends Lint.RuleWalker {
+    private isInJsxAttribute = false;
     private isInJsxExpression = false;
 
-    public visitJsxExpression(node: ts.JsxExpression) {
+    protected visitNode(node: ts.Node) {
+        if (nodeIsKind<ts.JsxAttribute>(node, ts.SyntaxKind.JsxAttribute)) {
```

Comment:
the type guard doesn't do much for us here... consider `node.kind === ...` for simplicity